x/liquidity/amm/pool.go

```diff
@@ -446,8 +331,8 @@ func (pool *RangedPool) BuyAmountTo(price sdk.Dec) (amt sdk.Int) {
 	if price.GTE(pool.Price()) {
 		return sdk.ZeroInt()
 	}
-	if pool.minPrice != nil && price.LT(*pool.minPrice) {
-		price = *pool.minPrice
+	if price.LT(pool.minPrice) {
+		price = pool.minPrice
 	}
 	sqrtXComp := utils.DecApproxSqrt(pool.xComp)
 	sqrtYComp := utils.DecApproxSqrt(pool.yComp)
```

Comment:
Calculated `dx` first because that amount is how much the pool can pay, after truncation(and I just found a mistake where there's no `.TruncateInt()` in `dx` calculation). `dy` should then be derived from `dx`.